### PR TITLE
Fixes multiprocessing jobs hanging when using remove_stripe_fw and retrieve_phase

### PR DESCRIPTION
**Pull request by lbluque**
I tested this a few times and had not hanging jobs. We might need to edit the `pyfftw.interfaces.cache.set_keepalive_time(5)` line or remove the caching altogether if hanging still occurs, but this seemed to work well. My idea is that the cache was being cleared before the multiprocessing jobs had been allocated and therefore hanging occured. Thanks to @HarinarayanKrishnan for catching this!

Also in this PR are some overdue unit tests for normalize_nf and find_center_pc

### CODE REVIEW

**Comment by dgursoy** [comment]
@dmpelt I remember you also experienced this and suggested a fix like this before. Any comments on this?

File: tomopy/prep/phase.py [hunk 1]
```diff
@@ -137,26 +138,20 @@ def retrieve_phase(
...
         proj = np.real(pyfftw.interfaces.numpy_fft.ifft2(
-            filtproj, planner_effort=_plan_effort(num_jobs))
+            filtproj, planner_effort=fftw_effort)
             ) / phase_filter.max()
         if pad:
             proj = proj[px:dy + px, py:dz + py]
         tomo[m] = proj
 
 
-def _plan_effort(num_jobs):
```
Comment (dgursoy) [inline tomopy/prep/phase.py]: Hi Luis - why did you remove this function? Isn't it safe to have it once and call if needed, e.g. phase retrieval and stripe removal?

Comment (lbluque) [inline tomopy/prep/phase.py]: I switched the function to a statement: `fftw_effort = 'FFTW_MEASURE' if num_jobs > 10 else 'FFTW_ESTIMATE'` (line 141 in phase.py and 132 in stripe.py).
This was to simplify the code (reduce function call and imports).
But you have a point that in terms of maintenance it is better to have it in one place.

**Comment by dmpelt** [comment]
I don't remember having problems with hanging in TomoPy due to `pyfftw` caching, but I did increase the timeout in some of my personal codes.

I am a bit confused why the code is hanging because of this: I would expect each process to have a separate cache, and that `pyfftw` would be able to run correctly even if the cache is cleaned. Maybe we can try the new `tomopy.set_debug` functionality of #196 to investigate why it is hanging? It might not hang in that case though.

Also, do we actually need to enable the cache? The documentation [here](https://hgomersall.github.io/pyFFTW/pyfftw/interfaces/interfaces.html#module-pyfftw.interfaces.cache) suggests that caching only helps when doing a large number of (very) small transforms, which might not be relevant for typical TomoPy usecases. Is it easy to perform some computation time tests to see whether caching actually has a significant impact? If not, simply disabling the cache might solve the hanging problem.

Slightly related: maybe it is also useful to generate and save wisdom files after doing a `'FFTW_ESTIMATE`' call. I can imagine that users will often perform FFTs of identical sizes (e.g. the detector size of their experimental setup), in which case the wisdom file will help to speedup the first FFT significantly.

**Comment by lbluque** [comment]
I agree with @dmpelt that we could disable the caching altogether. I did some quick comparisons caching vs non-chaching and speed differences were not very different. Plus disabling the cache also solves the hanging problem.

**Comment by xianghuix** [comment]
Is this the issue related to #120?


________________________________

**Comment by dmpelt** [comment]
@xianghuix yes, I think they could be related. In general, if something crashes in one of the child processes, the issue of #120 will occur. This is a bit difficult to fix fully, but in recent Python versions (>3) and with recent commits to TomoPy things should behave a bit better (i.e. crash and report an error instead of hanging).

**Comment by dgursoy** [comment]
What's the status of this PR? Should we still keep it open or is it already resolved?

File: tomopy/prep/phase.py (hunk 1 above)
Comment (lbluque) [inline tomopy/prep/phase.py]: @dgursoy should I switch this back to the original function call?

**Comment by lbluque** [comment]
Well, I think we should get rid of the pyfftw caching like @dmpelt mentioned.
In terms of #120, perhaps this [http://scikit-learn.org/stable/faq.html#why-do-i-sometime-get-a-crash-freeze-with-n-jobs-1-under-osx-or-linux](url) on scikit-image might be related.

**Comment by lbluque** [comment]
I removed pyfftw caching altogether.
@dgursoy Do you prefer to keep the `phase._plan_effort` function as it was before or keep the statement in this PR? This should be the last thing to address before merging.

**Comment by dgursoy** [comment]
I would keep ```_plan_effort``` function. Then the PR is ready to be merged. Thanks.

**Comment by lbluque** [comment]
Done.

**Comment by dgursoy** [comment]
Thanks. Any idea why the Travis is failing?

**Comment by lbluque** [comment]
Ah there was an error in the unit tests because I forgot to switch back the `_plan_effort` call in `_retrieve_phase`

**Comment by dgursoy** [comment]
All right, thanks!
